### ISSUE TITLE
bug: (YouTube/SponsorBlock) Change segment behaviour data imports incorrectly and when exported it's missing data

### ISSUE THREAD

**Issue by 1node0001** [inotia00/ReVanced_Extended#2404]
### Type

Error at runtime

### Tools used

RVX Manager

### Application

YouTube 19.16.39
Patches 4.14.1-dev.2

### Bug description

I have my Sponsorblock set like this:

Change segment behaviour

- Sponsor (skip automatically) - correctly imported
- Unpaid / Self Promotion (skip automatically) - incorrectly imported
- Interaction Reminder (skip automatically) - incorrectly imported
- Highlight (skip automatically) - incorrectly imported
- Intermission (skip automatically) - incorrectly imported
- Endcards (skip automatically) - incorrectly imported
- Preview (skip automatically) - incorrectly imported
- Filler Tangent (skip automatically) - incorrectly imported
- Music Non Section (skip automatically) - incorrectly imported

My Sponsorbock is set to Skip Automatically for every section above, and when I import settings only Sponsor is set to Skip Automatically. All the other sections are set wrong. Is this intended behaviour or is this a bug? Also when I looked at the exported text file, I saw that there was only 6 set to Skip, but there are 9 sections in total.

### Error logs

```shell
No available.
```

### Additional context

Settings exported (Sponsorblock)

"sb_filler": "skip",
"sb_filler_color": "#FFFFFF",
"sb_highlight": "skip",
"sb_highlight_color": "#FFFFFF",
"sb_interaction": "skip",
"sb_interaction_color": "#FFFFFF",
"sb_intro": "skip",
"sb_intro_color": "#FFFFFF",
"sb_local_time_saved_milliseconds": 76181,
"sb_local_time_saved_number_segments": 4,
"sb_music_offtopic": "skip",
"sb_music_offtopic_color": "#FFFFFF",
"sb_outro": "skip",
"sb_outro_color": "#FFFFFF",
"sb_preview": "skip",
"sb_preview_color": "#FFFFFF",
"sb_selfpromo_color": "#FFFFFF",
"sb_sponsor_color": "#FFFFFF",
"sb_toast_on_skip": false,
"sb_track_skip_count": false,
"sb_video_length_without_segments": true

### Device Environment

Android 14

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I didn't use any settings marked as `Experimental Flags`.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by inotia00** [comment]
This is not a bug

The explanation is a bit long, but if you have time, read the whole thing

There are two ways to import / export settings in RVX

One is `Import / Export settings` in SponsorBlock, and the other is `Import / Export settings` in Miscellaneous

`Import / Export settings` in SponsorBlock
====
![SB](https://github.com/user-attachments/assets/af7b6b9f-ce94-41cd-82eb-258df8a08fb3)

This is the settings for RVX as well as other platforms like SponsorBlock for web browsers (Chrome extension)

So it only contains settings related to SponsorBlock, and exports all settings that the user has not changed

(To match the Import / Export format of SponsorBlock on other platforms)

`Import / Export settings` in Miscellaneous
==
![Misc](https://github.com/user-attachments/assets/fbc3906c-e455-4ba7-9364-ba0a2f143a25)

This is the settings for RVX only, so it contains all RVX settings

There are about 400 settings used in RVX

Every time the RVX exports settings, if 400 settings are treated as text or file, unnecessary resources will be wasted

Therefore, `Import / Export setting` in Miscellaneous **ONLY EXPORTS SETTINGS MANUALLY CHANGED BY USERS**

In other words, some SponsorBlock settings that are not included in `Import / Export setting` of Miscellaneous are not the Segment behaviors that you changed manually

If you still don't understand, test the following:

1. Export the settings via `Import / Export setting` of Miscellaneous
2. Remove the app data of RVX
3. Import the settings via `Import / Export setting` of Miscellaneous
4. You can use the same settings
